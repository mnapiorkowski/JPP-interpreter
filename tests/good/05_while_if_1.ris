void main():
    let int i = -11;
    while (i < 10):
        i++;
        print i;
        if (i < 0):
            print ": negative";
            if (i % 2 == 0):
                println " even";
            else:
                println " odd";
            endif;
        elif (i > 0):
            print ": positive";
            if (i % 2 == 0):
                println " even";
            else:
                println " odd";
            endif;
        else:
            println ": zero";
        endif;
    endwhile;
    
    return;