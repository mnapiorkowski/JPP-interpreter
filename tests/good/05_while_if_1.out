-10: negative even
-9: negative odd
-8: negative even
-7: negative odd
-6: negative even
-5: negative odd
-4: negative even
-3: negative odd
-2: negative even
-1: negative odd
0: zero
1: positive odd
2: positive even
3: positive odd
4: positive even
5: positive odd
6: positive even
7: positive odd
8: positive even
9: positive odd
10: positive even
